### DISCUSSION TITLE
Questions about cheng models and Joint autoregressive models

### DISCUSSION THREAD

**Discussion by AlbertoPresta** [InterDigitalInc/CompressAI#208] [Q&A]
HI, 

Sorry if I bother you with other questions, which are the follwings:

- why did you use BufferedRansEncoder for the joint autoregressive model instead of the actual ransencoder? is it faster for decoding?

-  For what concers models in waseda.py, Did you implement the sum of Gaussians as explained in the reference papers or did you maintain the old entropy models (non-zeros Gaussians)? 

Thanks in advance for the answers,
Alberto

**Comment by YodaEmbedding** [comment]
1. No difference in the encoder performance since the buffer is immediately flushed after a single call anyways. Decoder performance will never be affected since the bitstream produced is identical.
2. A single Gaussian ($K=1$) entropy model is implemented in CompressAI. The paper suggests a $K=3$ Gaussian mixture. There isn't too much increase in RD performance with $K>1$, AFAIK. The primary gains in RD performance for cheng2020-anchor versus the previous model mbt2018 is almost entirely due to the modification in transforms $g_a, g_s$. The attention layers within the cheng2020-attn model do not appear to show significant differences in RD performance, either.

**Reply by AlbertoPresta** [reply]
ok thanks. So in theory it should be possible to use RansEncoder() as for the previous models right?

**Reply by YodaEmbedding** [reply]
Yes.
